src/contexts/wallet.context.tsx

```diff
@@ -0,0 +1,70 @@
+import { createContext, PropsWithChildren, useContext, useEffect } from 'react';
+import Web3 from 'web3';
+import { useApi } from '../hooks/api.hook';
+import { useSession } from '../hooks/session.hook';
+
+interface WalletInterface {
+  address?: string;
+  login: () => Promise<void>;
+  isInstalled: boolean;
+}
+
+const WalletContext = createContext<WalletInterface>(undefined as any);
+
+export function useWalletContext(): WalletInterface {
+  return useContext(WalletContext);
+}
+
+export function WalletContextProvider(props: PropsWithChildren): JSX.Element {
+  const { address, setAddress } = useSession();
+  const { getSignMessage } = useApi();
+  const { ethereum } = window as any;
+  const web3 = new Web3(Web3.givenProvider);
+
+  useEffect(() => {
+    web3.eth.getAccounts((err, accounts) => {
+      setAddress(verifyAccount(accounts));
+    });
+    ethereum.on('accountsChanged', (accounts: string[]) => {
+      setAddress(verifyAccount(accounts));
+    });
```

Comment:
What is this doing?

Reply:
You only need to connect once and you will stay connected with this site forever. The first function is there to read available (connected) accounts. The second function is to add a "listener" to the global ethereum object to receive updates, if the user changes his account in metamask.